consumer/consumererror/error.go

```diff
@@ -0,0 +1,178 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"errors"
+
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/collector/consumer/consumererror/internal/statusconversion"
+)
+
+// Error acts as a container for errors coming from pipeline components.
+// It may hold multiple errors from downstream components, and is designed
+// to act as a way to accumulate errors as it travels upstream in a pipeline.
+// `Error` should be obtained using `errors.As` and as a result, ideally
+// a single instance should exist in an error stack. If this is not possible,
+// the most `Error` object should be highest on the stack.
+//
+// Experimental: This API is at the early stage of development and may change without backward compatibility
+type Error struct {
+	errors []ErrorData
+}
+
+// ErrorData is intended to be used to encapsulate various information
```

Comment:
Would call this "Error" and the other "Errors"(or ErrorSlice)? 

Maybe "MultiErr" for the container? Not sure, but throwing ideas.

Reply:
I've renamed them to `ErrorContainer` (for the type that accumulates errors as they move upstream) and `Error` for the actual error with the data. What do you think?